packages/react-form/src/hooks/list/baselist.ts

```diff
@@ -62,10 +66,10 @@ export function useBaseList<Item extends object>(
   const [state, dispatch] = useListReducer(list);
 
   useEffect(() => {
-    if (!isEqual(list, state.initial)) {
+    if (!isEqual(list, state.initial) && !hasCleaned) {
```

Comment:
but perhaps we could remove state.initial from the dependency list there and avoid this hasCleaned state?

Reply:
I tried looking for other possibilities but the truth is, we will need a flag since if state.initial is changed, it will try and re-initialize the array. 

Perhaps we could do something better there by checking if state.initial is empty instead of using isEqual

Reply:
Figured out the solution. The solution they used on `Field.ts` was to remove the initial value from the dependency list as they explained there,

```
We actually do not want this to rerun when our `defaultValue` is updated. 
It can only happen independently of this callback when `newDefaultValue` is called by a user,
and we don't want to undue their hard work by resetting to `value`.
```

Reply:
Made the changes [here](https://github.com/Shopify/quilt/pull/1871/commits/e6e02a60dd00381bedafd67bd3b5e6df7b76c883) and added a test to make sure reinitialize is called when the list has changed.